icicle/backend/cpu/include/hash/blake2/blake2s.h

```diff
@@ -0,0 +1,116 @@
+
+#pragma once
+#include <stddef.h>
+#include <stdint.h>
+#include <string.h>
+#include "icicle/hash.h"
+
+using namespace icicle;
+namespace blake2s_cpu {
```

Comment:
All of Icicle types should be defined inside `icicle` namespace. So you need to have smth like

```
namespace icicle {
  namespace blake2s_cpu {
     ...
  }
}
```

But even so, I gave it a second thought on these BYTE, WORD, LONG. It's not worth for the users to use blakes2s through `icicle::blake2s_cpu::Blake2s` just because of these 3 typedefs. Maybe it's worth to just remove them. Another solution would be to move these typedefs into some namespace in utils.h or smth and then import them to your code wherever needed

e. g.
```
// utils.h
namespace icicle {
  namespace asm_types {
    typedef unsigned char BYTE;
    typedef unsigned int WORD;
    typedef unsigned long long LONG;
  }
}

// blake2s.h
namespace icicle {
  using namespace asm_types;
}
```